### PR TITLE
userrr

### PR DESCRIPTION
**Pull request by orangeOAO**

### CODE REVIEW

**Comment by nonorze-VOXX** [comment]
merge main first

File: backend/user_log/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+- 創建usertable
+
+    psql -U postgres -f UserInitializeTable.sql
```
Comment (nonorze-VOXX) [inline backend/user_log/README.md:3]: existed

File: backend/user_log/UserInitializeTable.sql [hunk 2]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE UserData(
+  ID serial PRIMARY KEY,
+  Name varchar(32) not null,
+  Password varchar(32) not null,
+  Email varchar(64) unique not null,
+  Address char(64) not null,
+  PhoneNumber varchar(16) not null,
+  Birthday DATE,
+  Authority int not null
+);
```
Comment (nonorze-VOXX) [inline backend/user_log/UserInitializeTable.sql]: existed
 SQL語法 建議要大寫

File: backend/user_log/user_log.go [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+		}
+
+		if strings.EqualFold(address, email) {
+			return errors.New("This Email has been registered")
+		}
+	}
+
+	return nil
+}
+func TF(tmp [4]bool) bool {
```
Comment (nonorze-VOXX) [inline backend/user_log/user_log.go]: titan fall?

File: backend/user_log/user_log.go [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+}
+
+// email格式
+func validMailAddress(address string) error {
+	_, err := mail.ParseAddress(address)
+	if err != nil {
+		return err
+	}
+	return nil
+}
```
Comment (nonorze-VOXX) [inline backend/user_log/user_log.go]: no need

File: backend/user_log/user_log.go [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+
+		notEnd := 0
+		for _, s := range spCode {
+			if i == s {
+				indNum[3] = 1
+				notEnd = 1
+				break
+			}
+		}
+
```
Comment (nonorze-VOXX) [inline backend/user_log/user_log.go]: maybe can it use 2D array?

**Review by nonorze-VOXX — CHANGES_REQUESTED** [review]

File: backend/user_log/user_log.go (hunk 4 above)
Comment (orangeOAO) [reply backend/user_log/user_log.go]: why

File: backend/user_log/UserInitializeTable.sql [hunk 6]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE IF NOT EXISTS user_datas(
+  user_id SERIAL PRIMARY KEY,
+  name VARCHAR(32) NOT NULL,
+  password VARCHAR(32) NOT NULL,
+  email VARCHAR(64) UNIQUE NOT NULL,
+  address VARCHAR(64) NOT NULL,
+  phone_number VARCHAR(16) NOT NULL,
+  birthday DATE,
+  authority INT NOT NULL
+);
```
Comment (nonorze-VOXX) [inline backend/user_log/UserInitializeTable.sql:10]: docker compose will not run this .sql , it's path is wrong
please check

File: backend/user_log/string/string.go [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+// const letterBytes = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+
+// // 生成随机字符串，n 代表生成长度
+// func Rand(n int) string {
+// 	b := make([]byte, n)
+// 	for i := range b {
+// 		b[i] = letterBytes[rand.Intn(len(letterBytes))]
+// 	}
+// 	return string(b)
+// }
```
Comment (nonorze-VOXX) [inline backend/user_log/string/string.go:14]: why have this file

Comment (orangeOAO) [reply backend/user_log/string/string.go:14]: 寫驗證ing

File: backend/user_log/UserInitializeTable.sql (hunk 6 above)
Comment (orangeOAO) [reply backend/user_log/UserInitializeTable.sql:10]: 這是測試用的之後會刪掉ouo

File: backend/user_log/string/string.go (hunk 7 above)
Comment (nonorze-VOXX) [reply backend/user_log/string/string.go:14]: 你要交在同一個pr？

Comment (orangeOAO) [reply backend/user_log/string/string.go:14]: 沒
其實我不知道為啥我後面push的會加到之前pr 的裡面

Comment (nonorze-VOXX) [reply backend/user_log/string/string.go:14]: 因為你在同一個branch上做
它的單位是branch 
你應該在要有一條做事 一條pr

Comment (orangeOAO) [reply backend/user_log/string/string.go:14]: 歐給歐給 回家改

**Pull request closed without merging**